### PR TITLE
Notification refactor

### PR DESCRIPTION
**Pull request by lsemesky**
this is the branch to add to hud and center.

### CODE REVIEW

File: app/js/components/notification/NotificationsModal.jsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    propTypes: {
+        backRoute: React.PropTypes.oneOfType([
+            React.PropTypes.string.isRequired,
+            React.PropTypes.func.isRequired
+        ])
+    },
+    
+    render: function() { 
+        return (
+            <Modal title="Notifications" ref="modal"
```
Comment (blynch11p) [inline app/js/components/notification/NotificationsModal.jsx]: This will be `modaltitle="Notifications"` with the recent updates to Modal.jsx.

**Comment by rkenyon1969** [comment]
Changing the following code may resolve "Add" button issue:

`"bookmark_notification_id": notification.id`
to
`"bookmark_notification_id": notification.notificationId`

**Comment by lsemesky** [comment]
@rkenyon1969  yes, I have already changed this on the local branch I made for Issue #658. Please feel free to take the ticket yourself if you want to keep working on it but be aware that I have already identified and addressed the issue and have been cleaning up the code before I submit a pr.

**Comment by rkenyon1969** [comment]
No. Please continue to work on this issue. Good job so far. While reviewing this and other PRs came across above solution and thought it might be helpful to your task. - thanks,

**Comment by blynch11p** [comment]
:+1: If needed along with #92?

**Comment by lsemesky** [comment]
closing because changes are integrated as a part of #92

**Pull request closed without merging**